src/widget/wmainmenubarbutton.cpp

```diff
@@ -0,0 +1,18 @@
+#include "widget/wmainmenubarbutton.h"
+#include "widget/wmainmenubar.h"
+
+WMainMenuBarButton::WMainMenuBarButton(QWidget* pParent, WMainMenuBar* pMainMenu)
+        : QPushButton("...", pParent),
+          WBaseWidget(this),
+          m_pMenu(nullptr) {
+    initialize(pMainMenu);
+}
+
+void WMainMenuBarButton::setup(const QDomNode& node, const SkinContext& context) {
+}
+
+void WMainMenuBarButton::initialize(WMainMenuBar* pMainMenu) {
```

Comment:
This function is not needed, the initialization is only performed once. The remaining 2 lines can be added to the constructor body.